### ISSUE TITLE
Build broken on compiler-cli @9.0.0-next.3

### ISSUE THREAD

**Issue by aleksandar-kandzhichki** [angular/angular#32272]
# 🐞 bug report

### Affected Package
@angular/compiler-cli@9.0.0-next.3

### Is this a regression?
Yes, it is a regression. 
Getting back to previous package version fixes the build (version 9.0.0-next.2 and 8.2.2 both work)

### Description
Updating package to latest (9.0.0-next.3) breaks the build of angular project.
The issue is the following: 
![image](https://user-images.githubusercontent.com/13217235/63532851-46d6a280-c514-11e9-8bf2-e56c86584a38.png)

## 🔬 Minimal Reproduction
https://github.com/aleksandar-kandzhichki/angular-compiler-cli-issue

I suppose using @angular/forms should reproduce the errors.
I tried with older versions of forms, core, compiler but it still failed to build until I got back the compiler-cli version

## 🔥 Exception or Error
ERROR in Error: Assertion error: failed to find directive with exportAs: ngForm
at '...\frontend\node_modules\@angular\compiler\bundles\compiler.umd.js:28409:31
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
@angular-devkit/architect         0.802.2
@angular-devkit/build-angular     0.802.2
@angular-devkit/build-optimizer   0.802.2
@angular-devkit/build-webpack     0.802.2
@angular-devkit/core              8.2.2
@angular-devkit/schematics        8.2.2
@angular/cdk                      8.1.3
@angular/cli                      8.2.2
@angular/flex-layout              7.0.0-beta.24
@angular/material                 8.1.3
@ngtools/webpack                  8.2.2
@schematics/angular               8.2.2
@schematics/update                0.802.2
rxjs                              6.5.2
typescript                        3.5.3
webpack                           4.38.0

</code></pre>

**Comment by atscott** [comment]
Hi Aleksandar, can you provide a minimal reproduction of this issue so we can investigate further? It's hard to determine what's happening with just the isolated stack trace. Thanks!

**Comment by aleksandar-kandzhichki** [comment]
Hello @atscott , does this do the job?
https://github.com/aleksandar-kandzhichki/angular-compiler-cli-issue

If yes, I will edit my post and add it there in the reproduction

**Comment by wojciechpolak** [comment]
This still happens with 9.0.0-next.6

**Comment by petterhoel** [comment]
Got same issue trying out a client project with `ng update @angular/cli@next @angular/core@next` today. 
```
ERROR in Error: Assertion error: failed to find directive with exportAs: ngForm
at C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28682:31                                                    
    at Array.forEach (<anonymous>)
    at DirectiveBinder.visitElementOrTemplate (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28665:29)           
    at DirectiveBinder.visitElement (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28639:76)                     
    at Element.visit (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:4160:71)                                     
    at C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28711:67                                                    
    at Array.forEach (<anonymous>)
    at DirectiveBinder.visitElementOrTemplate (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28711:27)           
    at DirectiveBinder.visitElement (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28639:76)                     
    at Element.visit (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:4160:71)                                     
    at C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28637:60                                                    
    at Array.forEach (<anonymous>)
    at DirectiveBinder.ingest (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28637:22)                           
    at Function.DirectiveBinder.apply (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28632:21)                   
    at R3TargetBinder.bind (C:\source\client\node_modules\@angular\compiler\bundles\compiler.umd.js:28479:38)                              
    at ComponentDecoratorHandler.typeCheck (C:\source\client\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\component.js:374:64)
m
** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
i ｢wdm｣: Failed to compile.
i ｢wdm｣: Compiling...

Date: 2019-10-10T15:22:16.469Z - Hash: 07d60cce74cb0d632ad2
                5 unchanged chunks

Time: 417ms
                   
ERROR in Cannot read property 'incrementalState' of null  
```

**Comment by Delagen** [comment]
The same error with RC

**Comment by petterhoel** [comment]
Issue still present for me as well with rc0. It works if I ` "aot": false` and `"buildOptimizer": false` or disable Ivy.

**Comment by petterhoel** [comment]
rc-2 seems to have resolved this for me 🥳

**Comment by JoostK** [comment]
@petterhoel Correct, this was fixed in #33454. As such, closing this issue.

**Comment by Delagen** [comment]
@JoostK Strange I can't run example latest rc
6
